### PR TITLE
Add new Faulted state for pod failures

### PR DESCRIPTION
**Pull request by ryescholin**

### CODE REVIEW

File: pod-operation/src/state_machine.rs [hunk 1]
```diff
@@ -183,6 +186,17 @@ impl StateMachine {
...
 
+	fn _enter_faulted(&mut self) {
+		info!("Entering Faulted state");
+		self.io
+			.of("/control-station")
+			.unwrap()
+			.emit("fault", "123")
+			.ok();
+		self.signal_light.disable();
+		self.brakes.engage();
```
Comment (taesungh) [inline pod-operation/src/state_machine.rs:197]: Chore: Following from #79 (once that is merged), we'll need to also disable the high-voltage system.

Comment (ryescholin) [reply pod-operation/src/state_machine.rs:197]: yuh

**Review by samderanova — APPROVED** [review]
Thanks for working on this! Please squash during merge once it's been approved by another.

**Review by taesungh — APPROVED** [review]
Thanks for adding, one note for future planning.

File: pod-operation/src/state_machine.rs [hunk 2]
```diff
@@ -187,6 +190,18 @@ impl StateMachine {
 		self.high_voltage_system.disable();
 	}
 
+	fn _enter_faulted(&mut self) {
+		info!("Entering Faulted state");
+		self.io
+			.of("/control-station")
+			.unwrap()
+			.emit("fault", "123")
+			.ok();
```
Comment (taesungh) [inline pod-operation/src/state_machine.rs:199]: Note: I think we'll eventually want to emit the reason of the fault meaning we'll have to move this to where the fault actually occurs, but we can revisit that as part of #56.

Comment (ryescholin) [reply pod-operation/src/state_machine.rs:199]: perf

**Pull request merged**